backend/src/nodes/utils/onnx_to_ncnn.py

```diff
@@ -3005,46 +3018,46 @@ def convert(self, is_fp16: bool = False, include_mem_data: bool = True):
                     layer.add_param(11, int(kernel_shape[0]))
 
                 if dilations.size == 1:
-                    layer.add_param(2, dilations[0])
+                    layer.add_param(2, int(dilations[0]))
                 elif dilations.size == 2:
-                    layer.add_param(2, dilations[1])
-                    layer.add_param(12, dilations[0])
+                    layer.add_param(2, int(dilations[1]))
+                    layer.add_param(12, int(dilations[0]))
 
                 if strides.size == 1:
-                    layer.add_param(3, strides[0])
+                    layer.add_param(3, int(strides[0]))
                 elif strides.size == 2:
-                    layer.add_param(3, strides[1])
-                    layer.add_param(13, strides[0])
+                    layer.add_param(3, int(strides[1]))
+                    layer.add_param(13, int(strides[0]))
 
                 if auto_pad == "SAME_UPPER":
                     layer.add_param(4, -233)
                 elif auto_pad == "SAME_LOWER":
                     layer.add_param(4, -234)
                 else:
                     if pads.size == 1:
-                        layer.add_param(4, pads[0])
+                        layer.add_param(4, int(pads[0]))
                     elif pads.size == 2:
-                        layer.add_param(4, pads[1])
-                        layer.add_param(14, pads[0])
+                        layer.add_param(4, int(pads[1]))
+                        layer.add_param(14, int(pads[0]))
```

Comment:
Lots of casting going on here. Why is that?

Reply:
Not having it gave me a bug when I added the checked cast in the optimizer iirc. I think it was that some of the values were np.int64 instead of int, which is now strictly required. Prior to checked_cast it didn't matter, because both types convert to a string the same.

Reply:
> it was that some of the values were np.int64 instead of an int

Argh. Guess we have no choice then.